src/main/DiagramApp.tsx

```diff
@@ -93,18 +93,20 @@ export default class DiagramApp extends React.Component<DiagramAppProps, Diagram
 	}
 
 	componentDidMount(): void {
+		let timeout = setTimeout(() => {
+			console.error(Locale[ProjectSettings.viewLanguage].authenticationTimeout);
+			this.handleChangeLoadingStatus(false, Locale[ProjectSettings.viewLanguage].authenticationError, true, false);
+		}, 15000);
 		keycloak.onTokenExpired = () => keycloak.updateToken(30).catch(() =>
 			this.handleChangeLoadingStatus(false, Locale[ProjectSettings.viewLanguage].authenticationExpired, true, false)
 		);
-		keycloak.onAuthError = (error) =>
-			console.error(error);
-		Promise.race([keycloak.init({onLoad: "check-sso", flow: "implicit"}),
-			new Promise((resolve, reject) => setTimeout(() => {
-				if (!keycloak.authenticated)
-					reject(console.error(Locale[ProjectSettings.viewLanguage].authenticationTimeout))
-			}, 15000))
-		])
+		keycloak.onAuthError = (error) => {
+			console.error(`${error.error}: ${error.error_description}`);
+			window.clearTimeout(timeout);
```

Comment:
This clearTimeout call may be redundant, the one that is run after init should take care of this use case as well. But then again it's Keycloak library, there may be a bug, so who knows :-)

Then again there is probably one use case that is not covered at all - if Keycloak init throws an error, then the code goes to the catch block, which should probably clear the timeout as well. But I'm only guessing here.

Reply:
In my testing, Keycloak goes to catch, but doesn't pass the error message there. Instead, it invokes onAuthError with the message as the argument (which seems odd to me, but maybe it is just a practice I am not aware the advantages of). So, the init either goes there, or into .then, which is where my decision to put it there too came from.